app/src/main/java/de/tum/in/tumcampusapp/component/other/generic/activity/ProgressActivity.kt

```diff
@@ -94,6 +103,41 @@ abstract class ProgressActivity(
         }
     }
 
+    protected fun showErrorSnackbar(t: Throwable) {
+        val messageResId = when (t) {
+            is UnknownHostException -> R.string.no_internet_connection
+            is InactiveTokenException -> R.string.error_access_token_inactive
+            is InvalidTokenException -> R.string.error_invalid_access_token
+            is MissingPermissionException -> R.string.error_no_rights_to_access_function
+            is TokenLimitReachedException -> R.string.error_access_token_limit_reached
+            is RequestLimitReachedException -> R.string.error_request_limit_reached
+            else -> R.string.error_unknown
+        }
+
+        showErrorSnackbar(messageResId)
+    }
+
+    protected fun showErrorSnackbar(messageResId: Int) {
+        runOnUiThread {
+            Snackbar.make(contentView, messageResId, Snackbar.LENGTH_LONG)
```

Comment:
When there's no internet connection, I would opt for a persistent snackbar ([LENGTH_INDEFINITE](https://developer.android.com/reference/android/support/design/widget/Snackbar#LENGTH_INDEFINITE))

Reply:
That would potentially obscure content. Additionally, the `Snackbars` currently can’t be swiped away – we need to introduce `CoordinatorLayouts` for that.